### DISCUSSION TITLE
midwayjs-bull 定时任务 导致服务器直接挂了

### DISCUSSION THREAD

**Discussion by mynamexjl** [midwayjs/midway#3705] [Q&A] [answered]
### Describe the problem（描述问题）

// 问题：
执行完定时任务后，有出现 未能捕获住的error 服务器会出现挂掉了，后重启，也出现过重启失败直接让两台服务直接不可用 
【两台ecs  如果是单台的话暂时没出现这个现象】 项目代码 均没有使用  .error 的方法。 麻烦帮忙查看下是什么问题造成的。redis的话还是之前那种魔改版的redis

midwayjs-bull  3.11.6 之前使用版本 均未出现这种情况
midwayjs-bull  新版出现了该问题

//  common-error 错误日志 ：
ERROR 5421 TypeError: Cannot read properties of undefined (reading 'error') (uncaughtException throw 1 times on pid:5421)
    at BullQueue.<anonymous> (/home/admin/auto-edd-autosdk-server/target/auto-edd-autosdk-server/node_modules/_@midwayjs_bull@3.15.2@@midwayjs/bull/dist/framework.js:81:29)
    at BullQueue.emit (node:events:525:35)
    at EventEmitter.emit (node:events:513:28)
    at EventEmitter.silentEmit (/home/admin/auto-edd-autosdk-server/target/auto-edd-autosdk-server/node_modules/_ioredis@5.3.2@ioredis/built/Redis.js:464:30)
    at Socket.<anonymous> (/home/admin/auto-edd-autosdk-server/target/auto-edd-autosdk-server/node_modules/_ioredis@5.3.2@ioredis/built/redis/event_handler.js:196:14)
    at Object.onceWrapper (node:events:628:26)
    at Socket.emit (node:events:525:35)
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
    
![image](https://github.com/midwayjs/midway/assets/25628679/78b6f4c8-3b6f-4b89-86a1-6a9cf00b9485)

    
// 代码 ：
@Processor(QUEUE_NAME.CRON, {
  repeat: {
    cron: FORMAT.CRONTAB.EVERY_HOUR,
  },
  removeOnComplete: CORN_REMOVE_ON_COMPLETE,
  removeOnFail: CORN_REMOVE_ON_FAIL,
})
export class CronProcessor implements IProcessor {
  @Logger()
  logger: ILogger;

  @Config('envType')
  env: string;

  async execute(data: any) {
    try {
      await this.resourceService.extractApiZipToFolder();
    } catch (error) {
      console.log(`corn process error: ${JSON.stringify(error)}`);
    }
  }
}    

### Midway Versions（Midway 版本）

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)
✓ @midwayjs/async-hooks-context-manager(3.15.0)
✓ @midwayjs/axios(3.15.2)
✓ @midwayjs/bootstrap(3.15.0)
✓ @midwayjs/bull(3.15.2)
✓ @midwayjs/bull-board(3.15.2)
✓ @midwayjs/cache-manager(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)
✓ @midwayjs/casbin-typeorm-adapter(not installed)
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.15.0)
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(3.15.2)
✓ @midwayjs/decorator(3.15.0)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(3.15.2)
✓ @midwayjs/info(not installed)
✓ @midwayjs/jwt(not installed)
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(not installed)
✓ @midwayjs/mongoose(3.15.2)
✓ @midwayjs/mqtt(not installed)
✓ @midwayjs/oss(3.15.2)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(3.15.2)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(3.15.2)
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(not installed)
✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(3.15.2)
✓ @midwayjs/swagger(3.15.2)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/typegoose(3.15.2)
✓ @midwayjs/typeorm(3.15.2)
✓ @midwayjs/upload(3.15.2)
✓ @midwayjs/validate(3.15.2)
✓ @midwayjs/version(not installed)
✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(3.15.2)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(not installed)
✓ @midwayjs/ws(not installed)
**********************************************************************
>> Check complete, all versions are healthy.
**********************************************************************

**Comment by czy88840616** [comment, marked as answer]
fixed in https://github.com/midwayjs/midway/pull/3707
